assembl/static2/js/app/components/form/multilingualTextFieldAdapter.jsx

```diff
@@ -0,0 +1,51 @@
+// @flow
+/*
+  Text field adapter for react-final-form that supports multilingual
+
+  value should have the following form:
+  { en: 'Hello', fr: 'Bonjour' }
+*/
+import React from 'react';
+import { type FieldRenderProps } from 'react-final-form';
+import { ControlLabel, FormGroup, FormControl, HelpBlock } from 'react-bootstrap';
+
+import { getValidationState } from './utils';
+
+type multilingualValue = { [string]: string };
+
+type Props = {
+  editLocale: string,
+  input: {
+    name: string,
+    onBlur: (?SyntheticFocusEvent<*>) => void,
+    onChange: (SyntheticInputEvent<*> | any) => void,
+    onFocus: (?SyntheticFocusEvent<*>) => void,
+    value: multilingualValue
+  },
+  label: string
+} & FieldRenderProps;
+
+const MultilingualTextFieldAdapter = ({
+  editLocale,
+  input: { name, onChange, value, ...otherListeners },
+  label,
+  meta: { error, touched },
+  ...rest
+}: Props) => {
+  const valueInLocale = value[editLocale] || '';
+  return (
+    <FormGroup validationState={getValidationState(error, touched)}>
+      {valueInLocale ? <ControlLabel htmlFor={name}>{label}</ControlLabel> : null}
+      <FormControl
```

Comment:
It's not clear here if you pass the id prop to FormControl that matches htmlFor you gave for ControlLabel.

Reply:
ok